publish-mqtt/src/main.rs

```diff
@@ -291,78 +374,92 @@ async fn disconnect_first_stale_sensor(
     homie: &mut HomieDevice,
     sensors_connected: &mut Vec<Sensor>,
     sensors_to_connect: &mut VecDeque<Sensor>,
-) -> Result<(), Box<dyn Error>> {
+) -> Result<(), anyhow::Error> {
     let now = Instant::now();
     if let Some(sensor_index) = sensors_connected
         .iter()
         .position(|s| now - s.last_update_timestamp > UPDATE_TIMEOUT)
     {
-        let sensor = sensors_connected.remove(sensor_index);
+        let mut sensor = sensors_connected.remove(sensor_index);
         println!(
             "No update from {} for {:?}, reconnecting",
             sensor.name,
             now - sensor.last_update_timestamp
         );
-        homie.remove_node(&sensor.node_id()).await?;
+        sensor.connection_status = ConnectionStatus::WatchdogTimeOut;
+        homie
+            .remove_node(&sensor.node_id())
+            .await
+            .with_context(|| std::line!().to_string())?;
         sensors_to_connect.push_back(sensor);
     }
     Ok(())
 }
 
 async fn service_bluetooth_event_queue(
-    bt_session: &BluetoothSession,
-    homie: &mut HomieDevice,
-    sensors_connected: &mut Vec<Sensor>,
-    sensors_to_connect: &mut VecDeque<Sensor>,
-) -> Result<(), Box<dyn Error>> {
+    state: Arc<Mutex<SensorState>>,
+    bt_session: &MijiaSession,
+) -> Result<(), anyhow::Error> {
+    println!("Subscribing to events");
+    let (msg_match, mut events) = bt_session.event_stream().await?;
+    println!("Processing events");
     // Process events until there are none available for the timeout.
```

Comment:
Is this comment still accurate? If so, we should be retrying before panicking, as we might legitimately not get any events for a while.

Reply:
Not accurate. Removed.